core-library/src/main/java/org/silverpeas/core/index/search/model/IndexSearcher.java

```diff
@@ -458,9 +486,11 @@ private MatchingIndexEntry createMatchingIndexEntry(ScoreDoc scoreDoc, String re
     if (topDocs != null) {
       for (int i = 0; i < topDocs.scoreDocs.length; i++) {
         ScoreDoc scoreDoc = topDocs.scoreDocs[i];
-        MatchingIndexEntry indexEntry = createMatchingIndexEntry(scoreDoc, query
-            .getRequestedLanguage(), searcher);
-        results.add(indexEntry);
+        final MatchingIndexEntry indexEntry =
+            createMatchingIndexEntry(scoreDoc, query.getRequestedLanguage(), searcher);
+        if (indexEntry != null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS2589)